ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,513 @@
+          Effect: Allow
+          Principal:
+            AWS: !Sub "${InstanceRole.Arn}"
+          Action:
+          - kms:DescribeKey
+          - kms:Encrypt
+          - kms:Decrypt
+          - kms:ReEncrypt*
+          - kms:GenerateDataKey
+          - kms:GenerateDataKeyWithoutPlaintext
+          Resource: '*'
+        - Sid: 'Default KMS Policy that enables IAM permissions'  # https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
+          Effect: 'Allow'
+          Principal:
+            AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+          Action:
+            - 'kms:*'
+          Resource: '*'
+        - Sid: "Allow administration of the key to CFN service role"
+          Effect: "Allow"
+          Principal:
+            AWS:
+              - !ImportValue
+                'Fn::Sub': '${AWS::Region}-sc-ec2vpc-launchrole-LaunchRoleArn'
+          Action:
+            - "kms:Create*"
+            - "kms:Describe*"
+            - "kms:Enable*"
+            - "kms:List*"
+            - "kms:Put*"
+            - "kms:Update*"
+            - "kms:Revoke*"
+            - "kms:Disable*"
+            - "kms:Get*"
+            - "kms:Delete*"
+            - "kms:ScheduleKeyDeletion"
+            - "kms:CancelKeyDeletion"
+          Resource: "*"
+
+  SSMParameterEncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      # This will need to be synced with the `packer-rstudio` project
+      AliasName: !Sub 'alias/service-catalog/synapse/cred/${LinuxInstance}'
+      TargetKeyId: !Ref SSMParameterEncryptionKey
+
+  LinuxInstance:
+    Type: AWS::EC2::Instance
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E1022
+            - W1011
+      'AWS::CloudFormation::Init':
+        configSets:
+          StartContainersJupyter:
+            - start_docker_network
+            - start_reverse_proxy
+            - make_work_dir
+            - start_jupyter_notebook
+          StartContainersRstudio:
+            - start_docker_network
+            - start_reverse_proxy
+            - make_work_dir
+            - start_rstudio_notebook
+        start_docker_network:
+          commands:
+            start_docker_network:
+              command: "docker network inspect ${NETWORK_NAME} ||  \
+                docker network create --driver bridge ${NETWORK_NAME}"
+              env:
+                NETWORK_NAME:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, DockerNetworkName]
+        start_reverse_proxy:
+          commands:
+            start_reverse_proxy:
+              command: "docker start reverse-proxy || \
+                sudo docker run -d -p 443:443 \
+                --name reverse-proxy \
+                --network ${NETWORK_NAME} \
+                --restart unless-stopped \
+                -e EC2_INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id) \
```

Comment:
Apparently we do not have to update.  I have tried this out in our dev Service Catalog and it works fine.

Reply:
it's because you are building on top of an older AWS AMI,  the [IMDSV1 will be removed in mid 2024](https://aws.amazon.com/blogs/aws/amazon-ec2-instance-metadata-service-imdsv2-by-default/).  We should either do this now or in a follow on PR.